static_frame/core/type_blocks.py

```diff
@@ -531,6 +532,17 @@ def copy(self) -> 'TypeBlocks':
         '''
         return self.__copy__()
 
+    def __sizeof__(self: 'TypeBlocks') -> int:
+        # TODO: Make sure these don't have duplicates
+        return sum(getsizeof(o) for o in [
+            self._blocks,
+            *self._blocks,
+            self._dtypes,
+            self._index,
+            self._shape,
+            self._row_dtype
+        ])
```

Comment:
Also, pro-tip, this could be even slicker by using `map`.
```suggestion
        return sum(map(getsizeof, (
            self._blocks,
            *self._blocks,
            self._dtypes,
            self._index,
            self._shape,
            self._row_dtype
        ))
```
Same for the other methods

Reply:
I like to lean toward comprehensions personally. Either way, I changed this to a function and unfortunately had to get rid of the comprehension to check for duplicated ids.

Do we generally try to use tuples instead of lists when the intention is just to create an iterable?

Reply:
I only create lists if I need the ability to grow. In this case, we just building up a temporary container to feed its values into a function repeatedly, and so the best choice is a `tuple`, plus, `tuple` is a bit smaller than a `list`.

If you find the `map` syntax ugly, you don't need to change it, but I would recommend using a `tuple` instead of `list`. But just for fun, calling `map` is faster in this case than using a comprehension.

Performance:
```python
14:38:48  In [8] %timeit sum(sys.getsizeof(o) for o in l)
2.3 ms ± 195 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

14:38:53  In [9] %timeit sum(map(sys.getsizeof, l))
1.65 ms ± 66.3 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Memory:
```python
14:39:49  In [10] len(l)
14:40:32 Out [10] 10000

14:39:44  In [11] sys.getsizeof(t)
14:39:47 Out [11] 80040

14:39:47  In [12] sys.getsizeof(l)
14:39:49 Out [12] 87616
```

These are all pretty small optimizations, so they are not crucial, I just think it's worth pointing out the cost/benefit analysis of the different approaches :)

Reply:
I'm on-board with using a tuple in cases where things don't need to grow. Interesting to see the performance benefit of map. I am still not a big fan of the map syntax so I plan to leave in the comprehensions.